src/filesystem/drive/directory_handle.rs

```diff
@@ -519,31 +519,41 @@ impl DirectoryHandle {
 
         drop(inner_read);
 
-        let (parent_path, name) = path.split_at(path.len() - 1);
-        let file_name = NodeName::try_from(name[0]).map_err(OperationError::InvalidName)?;
+        let existing_file = match walk_path(&self.inner, self.cwd_id, path, 0).await {
+            Ok(WalkState::FoundNode { node_id }) => Some(node_id),
+            _ => None,
+        };
 
-        tracing::info!(?path, ?file_name, "drive::write");
+        let data_size = data.len() as u64;
+        let new_permanent_id = if let Some(existing_file) = existing_file {
```

Comment:
It's probably worth bumping this up into the match. If it find a file it could convert it into a permanent ID immediately to be used later on. In any other case from the walk_path we should then attempt the creation of a new Permanent ID and follow our normal error handling for missing / paths and the like.

Reply:
> It's probably worth bumping this up into the match.

Did just that and compiler told me it's used outside of the match on line 577 and then again on line 687. I don't mind a bit of duplication, but I don't know what's more important to use locality or lack of duplication.